DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vrrp/test_vrrp_advert.py

```diff
@@ -0,0 +1,138 @@
+from operator import itemgetter
+from math import isclose as is_close
+import asyncio
+import pytest
+
+from dent_os_testbed.lib.tc.tc_filter import TcFilter
+from dent_os_testbed.lib.tc.tc_qdisc import TcQdisc
+
+from dent_os_testbed.test.test_suite.functional.vrrp.vrrp_utils import (
+    setup_topo_for_vrrp,
+)
+
+from dent_os_testbed.utils.test_utils.tc_flower_utils import (
+    tcutil_get_tc_stats_pref_map,
+)
+
+
+pytestmark = [
+    pytest.mark.suite_functional_vrrp,
+    pytest.mark.usefixtures('cleanup_ip_addrs', 'enable_ipv4_forwarding', 'cleanup_bridges'),
+    pytest.mark.asyncio,
+]
+
+
+@pytest.mark.usefixtures('cleanup_qdiscs')
+async def test_vrrp_advert_interval(testbed, configure_vrrp):
+    """
+    Test Name: test_vrrp_advert_interval
+    Test Suite: suite_functional_vrrp
+    Test Overview:
+        Verify VRRP advertisement interval setting
+    Test Procedure:
+    1. Configure aggregation router
+    2. Configure infra devices
+    3. Configure VRRP on infra devices
+    4. Capture VRRP advertisement packets from each DUT for X seconds
+    5. Verify X packets are received
+    6. Change advertisement interval to 0.1 second
+    7. Verify 10*X packets are received
+
+    Setup:
+            agg
+         ___| |___
+      L0 |       | L1
+         |       |
+    infra[0]    infra[1]
+
+    Configure:
+        agg:
+            L0 and L1: master bridge
+            bridge: ip address 192.168.1.5/24
+        infra[0]:
+            L0 (port/bridge):
+                ip address 192.168.1.3/24
+                vrrp 40 ip 192.168.1.2 prio 200
+        infra[1]:
+            L1 (port/bridge):
+                ip address 192.168.1.4/24
+                vrrp 40 ip 192.168.1.2 prio 100
+    """
+    sleep_time_s = 30
+    vrrp_ip = '192.168.1.2'
+    vr_id = 40
+    pref = 100
+    tolerance = 0.1  # 10%
+
+    # 1. Configure aggregation router
+    # 2. Configure infra devices
+    config = await setup_topo_for_vrrp(testbed, use_bridge=True)
+    infra, agg, bridge, links = itemgetter('infra', 'agg', 'bridge', 'links')(config)
+    links = [link for link in links if agg in link]
+
+    # 3. Configure VRRP on infra devices
+    vrrp_ifaces = [bridge, bridge]
+    advert_int = 1
+    await asyncio.gather(*[
+        configure_vrrp(dent, state=state, prio=prio, vr_ip=vrrp_ip, vr_id=vr_id,
+                       dev=port, additional_opts=[f'advert_int {advert_int}'])
+        for dent, port, state, prio
+        in zip(infra, vrrp_ifaces, ['MASTER', 'BACKUP'], [200, 100])])
+    await asyncio.sleep(10)  # wait for keepalived to start
```

Comment:
use variable for sleep? 
wait for keepalived to start  on 81, 121 lines

Reply:
done